lib/bone.js

```diff
@@ -364,6 +364,11 @@ class Bone {
     for (const name in attributes) {
       const value = this.attribute(name);
       if (value != null) data[name] = value;
+      // set defaultValues
+      const { defaultValue } = attributes[name];
+      if (value === undefined && defaultValue != null) {
+        this[name] = defaultValue;
```

Comment:
should be `data[name] = defaultValue` here

L368-L371 can be wrapped in the else branch of `if (value != value)` or vice versa